src/Upgrade/Infrastructure/PackageManager/CommandExecutor/ComposerCommandExecutor.php

```diff
@@ -147,6 +170,11 @@ protected function getPackageString(PackageCollection $packageCollection): strin
     {
         $result = '';
         foreach ($packageCollection->toArray() as $package) {
+            if ($package->getVersion() === '') {
+                $result = sprintf('%s %s', $result, $package->getName());
```

Comment:
Only the last result is needed, right?

Reply:
Why last? we have different formats for packages with versions and without one. probably you confused with break;